tests/nixos/fetch-git/test-cases/lfs/default.nix

```diff
@@ -0,0 +1,202 @@
+
+    assert int(file_size_bad_lfs_without_lfs) >= expected_max_size_lfs_pointer, \
+      f"non lfs-enrolled file is {file_size_bad_lfs_without_lfs}b (<{expected_max_size_lfs_pointer}b), probably a test implementation error"
+    client.succeed(f"cmp {repo.path}/black_sheep {fetched_bad_lfs_without_lfs}/black_sheep >&2")
+
+    # finally fetch with lfs=true, and check that the bad file does not break anything
+    fetchGit_bad_lfs_with_lfs_expr = f"""
+      builtins.fetchGit {{
+        url = "{repo.remote}";
+        rev = "{bad_lfs_rev}";
+        ref = "main";
+        lfs = true;
+      }}
+    """
+    fetchGit_bad_lfs_with_lfs = client.succeed(f"""
+      nix eval --impure --raw --expr '({fetchGit_bad_lfs_with_lfs_expr}).outPath'
+    """)
+
+    client.succeed(f"cmp {repo.path}/black_sheep {fetchGit_bad_lfs_with_lfs}/black_sheep >&2")
+
+
+    # Add an lfs-enrolled file to the repo
+    ############################################################################
+
+    client.succeed(f"dd if=/dev/urandom of={repo.path}/beeg bs=1M count=1 >&2")
+    client.succeed(f"{repo.git} lfs install >&2")
+    client.succeed(f"{repo.git} lfs track --filename beeg >&2")
+    client.succeed(f"{repo.git} add : >&2")
+    client.succeed(f"{repo.git} commit -m 'add lfs file' >&2")
+    client.succeed(f"{repo.git} push origin main >&2")
+
+    # memorize the revision
+    lfs_file_rev = client.succeed(f"{repo.git} rev-parse HEAD").strip()
+
+    # first fetch without lfs, check that we did not smudge the file
+    fetchGit_nolfs_expr = f"""
+      builtins.fetchGit {{
+        url = "{repo.remote}";
+        rev = "{lfs_file_rev}";
+        ref = "main";
+        lfs = false;
+      }}
+    """
+    fetched_nolfs = client.succeed(f"""
+      nix eval --impure --raw --expr '({fetchGit_nolfs_expr}).outPath'
+    """)
+
+    # check that file was not smudged
+    file_size_nolfs = client.succeed(f"stat -c %s {fetched_nolfs}/beeg").strip()
+
+    assert int(file_size_nolfs) < expected_max_size_lfs_pointer, \
+      f"did not set lfs=true, yet lfs-enrolled file is {file_size_nolfs}b (>{expected_max_size_lfs_pointer}b), probably smudged when we should not have"
+
+    # now fetch with lfs=true and check that the file was smudged
+    fetchGit_lfs_expr = f"""
+      builtins.fetchGit {{
+        url = "{repo.remote}";
+        rev = "{lfs_file_rev}";
+        ref = "main";
+        lfs = true;
+      }}
+    """
+    fetched_lfs = client.succeed(f"""
+      nix eval --impure --raw --expr '({fetchGit_lfs_expr}).outPath'
+    """)
+
+    assert fetched_lfs != fetched_nolfs, \
+      f"fetching with and without lfs yielded the same store path {fetched_lfs}, fingerprinting error?"
+
+    # check that file was smudged
+    file_size_lfs = client.succeed(f"stat -c %s {fetched_lfs}/beeg").strip()
+    assert int(file_size_lfs) >= expected_max_size_lfs_pointer, \
```

Comment:
This was a conceptual test, since a pointer cannot be >= 1024 by definition, can definitely also check for the full 1MiB size. Will rewrite this with the other test changes when I have the time